x-pack/solutions/security/packages/expandable-flyout/src/hooks/use_window_width.ts

```diff
@@ -9,18 +9,150 @@ import { useLayoutEffect, useState } from 'react';
 import { useDispatch } from '../store/redux';
 import { setDefaultWidthsAction } from '../store/actions';
 
-export const RIGHT_SECTION_MIN_WIDTH = 380;
-export const MIN_RESOLUTION_BREAKPOINT = 992;
-export const RIGHT_SECTION_MAX_WIDTH = 750;
-export const MAX_RESOLUTION_BREAKPOINT = 1920;
+export const RESOLUTION_BREAKPOINTS = {
+  RIGHT_SECTION: {
+    MIN: 992, // resolution below which the width is fixed to its SECTION_WIDTHS.RIGHT.MIN value
+    OVERLAY_MAX: 1920, // resolution above which the overlay width is fixed to its SECTION_WIDTHS.RIGHT.MAX_OVERLAY value
+    PUSH_MIN: 1600, // resolution below which the push width is fixed to its SECTION_WIDTHS.RIGHT.MIN value
+    PUSH_MAX: 2560, // resolution above which the push width is fixed to its SECTION_WIDTHS.RIGHT.MAX_PUSH value
+  },
+  LEFT_SECTION: {
+    MIN: 1600, // resolution below which the overlay width goes full width (minus the padding) and the push width goes to its fixed SECTION_WIDTHS.LEFT.PUSH.MIN value
+  },
+};
+export const FULL_WIDTH_PADDING = 48;
+export const NAVIGATION_WIDTH = 200;
 
-const LEFT_SECTION_MAX_WIDTH = 1500;
+export const SECTION_WIDTHS = {
+  RIGHT: {
+    MIN: 380,
+    MAX_OVERLAY: 750,
+    MAX_PUSH: 600,
+  },
+  LEFT: {
+    OVERLAY: {
+      MAX: 1500,
+    },
+    PUSH: {
+      MIN: 380,
+      MAX: 1000,
+    },
+  },
+};
 
-const FULL_WIDTH_BREAKPOINT = 1600;
-export const FULL_WIDTH_PADDING = 48;
+/**
+ * Calculates the default widths for the right section of the expandable flyout in push and overlay modes.
+ *
+ * For overlay mode, the flyout right section scales as follows:
+ *  - for window widths below 380px, we make sure that the width is identical to the window width
```

Comment:
very minor nit for my single cell brain, this looks cleaner for me, wdyt?
```
 *  - for window widths below 380px, we make sure that the width is identical to the window width
 *  - for window widths 380px to 992px, the width is fixed at 380px
 *  - for window widths 992px to 1920px, the width scales linearly between 380px and 750px
 *  - for window widths above 1920px, the width is fixed at 750px
```

Reply:
what do you think about referencing the name of the number (Right panel min) versus actual number? for one, if the number is changed in the future, the dev needs to remember to change all the comments

or vice versa - use numbers in the function directly and matches the comments?